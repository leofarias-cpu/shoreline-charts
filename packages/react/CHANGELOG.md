# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.1.0-labs.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.1...@vtex/onda-react@0.1.0-labs.2) (2021-05-13)


### Features

* **onda-react:** allow state property ([47bfb26](https://github.com/vtex/onda/commit/47bfb26bdb2f2f5496a9458505913f8133dfbee2))





# [0.1.0-labs.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.0...@vtex/onda-react@0.1.0-labs.1) (2021-05-13)

**Note:** Version bump only for package @vtex/onda-react





# 0.1.0-labs.0 (2021-05-13)


### Features

* **onda:** add /react package ([4728b63](https://github.com/vtex/onda/commit/4728b631011945c4bd25a6e2de57c2d2e4f52c17))
