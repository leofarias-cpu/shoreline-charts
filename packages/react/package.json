{
  "name": "@vtex/admin-ui-react",
  "license": "MIT",
  "version": "0.0.0",
  "main": "dist/index.js",
  "module": "dist/admin-ui-react.esm.js",
  "typings": "dist/index.d.ts",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist",
    "src"
  ],
  "engines": {
    "node": ">=10"
  },
  "repository": {
    "directory": "packages/react",
    "type": "git",
    "url": "git+https://github.com/vtex/onda.git"
  },
  "bugs": {
    "url": "https://github.com/vtex/onda/issues"
  },
  "scripts": {
    "start": "tsdx watch --verbose --noClean",
    "build": "tsdx build --tsconfig tsconfig.json",
    "build:fast": "tsdx build --tsconfig tsconfig.json --transpileOnly --format esm,cjs",
    "test": "tsdx test"
  },
  "peerDependencies": {
    "@emotion/css": ">=11",
    "@emotion/react": ">=11",
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@vtex/onda-core": "^0.1.0-labs.8",
    "polished": "^4.1.2"
  },
  "dependencies": {
    "@emotion/is-prop-valid": "^1.1.0",
    "@vtex/onda-util": "^0.1.0-labs.3",
    "react-is": "^17.0.2",
    "tiny-invariant": "^1.1.0",
    "tiny-warning": "^1.0.3"
  }
}
