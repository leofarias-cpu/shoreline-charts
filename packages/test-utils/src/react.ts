import {
  render,
  renderHook,
  act,
  screen,
  fireEvent,
} from '@testing-library/react'

export { render, renderHook, act, screen, fireEvent }
