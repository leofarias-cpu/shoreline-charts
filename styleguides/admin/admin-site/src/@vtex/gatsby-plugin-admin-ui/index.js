export default {
  sizes: {
    header: '4.5rem',
    content: '90rem',
  },
  space: {
    content: '4.5rem',
  },
}
