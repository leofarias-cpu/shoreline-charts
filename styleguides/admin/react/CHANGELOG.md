# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.4.3](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.2...@vtex/onda-react@0.4.3) (2021-07-20)

**Note:** Version bump only for package @vtex/onda-react





## [0.4.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.1...@vtex/onda-react@0.4.2) (2021-07-19)

**Note:** Version bump only for package @vtex/onda-react





## [0.4.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.0...@vtex/onda-react@0.4.1) (2021-07-16)


### Bug Fixes

* **onda-react:** remove the cleaning of options properties from jsx ([b8ec31c](https://github.com/vtex/onda/commit/b8ec31cf1e557996ea8eb7027edbcfcbe55845fb))





# [0.4.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.3.0...@vtex/onda-react@0.4.0) (2021-07-15)


### Features

* **admin-ui/ultimate-table:** support sorting ([7453750](https://github.com/vtex/onda/commit/745375001bd28bb73e8a229f14c12e78bd0385da))





# [0.3.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.2...@vtex/onda-react@0.3.0) (2021-07-15)


### Bug Fixes

* **onda-react:** jsx refs type inference ([5a24609](https://github.com/vtex/onda/commit/5a2460960c419a272f16d76a43f9575247643f2a))
* **onda-react:** jsx rendering issues ([6fbe2b5](https://github.com/vtex/onda/commit/6fbe2b585ab083a99a1aa806de3ed0e23ac2451f))
* **onda-react:** typescript inference for jsx.elements ([fb9e365](https://github.com/vtex/onda/commit/fb9e365df6a877a274bc3ea26131a4015fbbcd33))


### Features

* **onda-react:** add ComponentProps type ([e13433c](https://github.com/vtex/onda/commit/e13433c2b9611de3cd6723541fe82064bd1f10cb))





## [0.2.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.1...@vtex/onda-react@0.2.2) (2021-07-05)

**Note:** Version bump only for package @vtex/onda-react





## [0.2.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.0...@vtex/onda-react@0.2.1) (2021-07-01)


### Bug Fixes

* **onda-core:** build imports ([9d30f9e](https://github.com/vtex/onda/commit/9d30f9e743d75b03995547e4d3052aa5a510e095))





# [0.2.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.1...@vtex/onda-react@0.2.0) (2021-07-01)


### Bug Fixes

* **onda-react:** remove unnecessary state ([1465e69](https://github.com/vtex/onda/commit/1465e695a6cb9be6ba3773d0acf51194a0713ea2))
* **onda-react:** resolve jsx hof types ([2e185ac](https://github.com/vtex/onda/commit/2e185ac2dd319f30d1eb4332a13a2299a38e06cb))


### Features

* **onda-react:** rephrase createComponent as jsx ([dd157ff](https://github.com/vtex/onda/commit/dd157ffa2500ed8e911b7d2d04db65268fc7f27b))





## [0.1.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.4...@vtex/onda-react@0.1.1) (2021-06-29)


### Bug Fixes

* **onda-react:** package.json version ([fb98ac4](https://github.com/vtex/onda/commit/fb98ac411f80ae253225940a7c687cb458677e4a))





# 0.1.0-labs.4 (2021-06-24)


### Features

* **onda-packages:** add onda packages ([b619064](https://github.com/vtex/onda/commit/b619064d04d190d5615a2832dbcb86d96efd16ee))





# [0.1.0-labs.3](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.2...@vtex/onda-react@0.1.0-labs.3) (2021-05-19)


### Features

* **onda:** allow global styles from theme ([aff3fa4](https://github.com/vtex/onda/commit/aff3fa449c6b92004312bf8ee4f648d7e6258dc0))
* **onda-core:** transform theme values on css variables ([3a8a379](https://github.com/vtex/onda/commit/3a8a379c3ccebf4738b65d39fc38aa8ac594e621))
* **onda-runtime-emotion:** allow default entries ([73a0dea](https://github.com/vtex/onda/commit/73a0deafda0b18ca09ca6eca00637d0c976055d8))





# [0.1.0-labs.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.1...@vtex/onda-react@0.1.0-labs.2) (2021-05-13)


### Features

* **onda-react:** allow state property ([47bfb26](https://github.com/vtex/onda/commit/47bfb26bdb2f2f5496a9458505913f8133dfbee2))





# [0.1.0-labs.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.0...@vtex/onda-react@0.1.0-labs.1) (2021-05-13)

**Note:** Version bump only for package @vtex/onda-react





# 0.1.0-labs.0 (2021-05-13)


### Features

* **onda:** add /react package ([4728b63](https://github.com/vtex/onda/commit/4728b631011945c4bd25a6e2de57c2d2e4f52c17))
