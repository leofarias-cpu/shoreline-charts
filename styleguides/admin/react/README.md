# Onda component model

## jsx

You can use the `jsx` function to create new components using elements or components as reference. It generates an `OndaJSXComponent` that inherits and enhances the reference types and styles.

### Referencing elements

To reference elements, simply pass its string representation to the function.

```jsx
const BlueSquare = jsx('div')({
  bg: 'blue',
  size: 50,
})

render(<BlueSquare />)
```

#### Aliasing

`jsx` is aliased with any valid `JSX.Element` for faster typing.

```jsx
const BlueSquare = jsx.div({
  bg: 'blue',
  size: 50,
})

render(<BlueSquare />)
```

### Referencing components

The idea is the same as elements. It will inherit the types and behavior of the passed component.

```jsx
import { motion } from 'framer-motion'

const Scale = jsx(motion.div)({
  bg: 'blue',
  size: 50,
})

render(<Scale animate={{ scale: 2 }} transition={{ duration: 0.5 }} />)
```
