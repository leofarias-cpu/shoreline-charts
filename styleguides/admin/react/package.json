{
  "name": "@vtex/onda-react",
  "license": "MIT",
  "version": "0.1.0",
  "main": "dist/index.js",
  "module": "dist/onda-react.esm.js",
  "typings": "dist/index.d.ts",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist",
    "src"
  ],
  "engines": {
    "node": ">=10"
  },
  "repository": {
    "directory": "packages/react",
    "type": "git",
    "url": "git+https://github.com/vtex/onda.git"
  },
  "bugs": {
    "url": "https://github.com/vtex/onda/issues"
  },
  "scripts": {
    "start": "tsdx watch --verbose --noClean",
    "build": "tsdx build --tsconfig tsconfig.json",
    "build:fast": "tsdx build --tsconfig tsconfig.json --transpileOnly --format esm,cjs",
    "test": "tsdx test --passWithNoTests",
    "storybook": "start-storybook -p 6005",
    "storybook:build": "yarn --cwd ../../ build && yarn run build-storybook"
  },
  "peerDependencies": {
    "@emotion/css": ">=11",
    "@emotion/react": ">=11",
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@vtex/admin-core": "^0.10.10",
    "@vtex/onda-util": "^0.1.0-labs.2",
    "polished": "^4.1.2"
  },
  "dependencies": {
    "@emotion/is-prop-valid": "^1.1.0",
    "react-is": "^17.0.2",
    "tiny-invariant": "^1.1.0",
    "tiny-warning": "^1.0.3"
  }
}
