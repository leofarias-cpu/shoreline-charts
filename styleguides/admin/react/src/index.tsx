export * from './tag'
export * from './domElements'
export * from './jsx'
export * from './types'
export * from './useStyleSheet'
export * from './util'
export * from './symbols'
