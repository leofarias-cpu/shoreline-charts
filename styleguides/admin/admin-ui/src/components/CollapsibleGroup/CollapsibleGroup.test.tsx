import React from 'react'
import { render } from '@testing-library/react'

import { CollapsibleGroup } from './index'
import { ThemeProvider } from '@vtex/admin-core'
import {
  useCollapsible,
  Collapsible,
  DisclosureStateReturn,
} from '../Collapsible'

function CollapsibleState({
  children,
}: {
  children: (state: DisclosureStateReturn) => JSX.Element
}) {
  const state = useCollapsible({ baseId: 'mocked-id' })

  return children(state)
}

describe('CollapsibleGroup tests', () => {
  it('should have overridable styles', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <CollapsibleGroup
          data-testid="collapsible-group"
          styleOverrides={{ bg: 'azure' }}
        >
          <CollapsibleState>
            {(state) => (
              <Collapsible state={state}>
                <Collapsible.Header label="Collapsible Header" />
                <Collapsible.Content>Collapsible Content</Collapsible.Content>
              </Collapsible>
            )}
          </CollapsibleState>
        </CollapsibleGroup>
      </ThemeProvider>
    )

    expect(getByTestId('collapsible-group')).toHaveStyleRule(
      'background-color',
      'azure'
    )
  })

  it('should match snapshot', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <CollapsibleGroup>
          <CollapsibleState>
            {(state) => (
              <Collapsible state={state}>
                <Collapsible.Header label="Collapsible Header" />
                <Collapsible.Content>Collapsible Content</Collapsible.Content>
              </Collapsible>
            )}
          </CollapsibleState>
        </CollapsibleGroup>
      </ThemeProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should not render invalid children', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <CollapsibleGroup>
          {undefined}
          <CollapsibleState>
            {(state) => (
              <Collapsible state={state}>
                <Collapsible.Header label="Collapsible Header" />
                <Collapsible.Content>Collapsible Content</Collapsible.Content>
              </Collapsible>
            )}
          </CollapsibleState>
          {null}
        </CollapsibleGroup>
      </ThemeProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
