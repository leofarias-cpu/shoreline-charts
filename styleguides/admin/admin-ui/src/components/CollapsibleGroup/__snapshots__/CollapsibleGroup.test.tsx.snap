// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleGroup tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-5 {
  background-color: #FFFFFF;
  border-radius: 4px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 1.5rem;
  padding-left: 0.5rem;
}

.emotion-4 div>button:nth-of-type(n+2) {
  margin-left: 0.25rem;
}

.emotion-2 {
  font-family: VTEXTrustVF,-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'wght' 92;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  text-transform: capitalize;
  background-color: transparent;
  color: #323845;
  font-size: 0.875rem;
  height: 40px;
  width: auto;
  padding-left: 0.5rem;
  padding-right: 0.75rem;
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-2:hover {
  color: #214491;
  background-color: #cad7f1;
}

.emotion-2:active {
  color: #1a3470;
  background-color: #bacbec;
}

.emotion-2:disabled {
  color: #898F9E;
}

.emotion-2 svg {
  width: 24px;
  min-width: 24px;
  height: 24px;
  min-height: 24px;
  margin: 0.25rem;
}

.emotion-2:hover {
  background-color: transparent;
}

.emotion-2:active {
  background-color: transparent;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-6 {
  background-color: #FFFFFF;
  color: #323845;
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  border-color: #E0E2E7;
  padding: 0rem;
}

.emotion-0 {
  min-width: 24px;
  min-height: 24px;
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(90deg);
  -moz-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition: -webkit-transform 150ms ease;
  transition: transform 150ms ease;
}

<div
    class="emotion-6"
  >
    <div
      class="emotion-5"
    >
      <header
        class="emotion-4"
      >
        <button
          aria-controls="mocked-id"
          aria-expanded="false"
          class="emotion-2"
          type="button"
        >
          <div
            class="emotion-1"
          >
            <svg
              class="emotion-0"
              fill="none"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 14L12 10L8 14"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
            Collapsible Header
          </div>
        </button>
        <div
          class="emotion-3"
        />
      </header>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CollapsibleGroup tests should not render invalid children 1`] = `
<DocumentFragment>
  .emotion-5 {
  background-color: #FFFFFF;
  border-radius: 4px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 1.5rem;
  padding-left: 0.5rem;
}

.emotion-4 div>button:nth-of-type(n+2) {
  margin-left: 0.25rem;
}

.emotion-2 {
  font-family: VTEXTrustVF,-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'wght' 92;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  text-transform: capitalize;
  background-color: transparent;
  color: #323845;
  font-size: 0.875rem;
  height: 40px;
  width: auto;
  padding-left: 0.5rem;
  padding-right: 0.75rem;
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-2:hover {
  color: #214491;
  background-color: #cad7f1;
}

.emotion-2:active {
  color: #1a3470;
  background-color: #bacbec;
}

.emotion-2:disabled {
  color: #898F9E;
}

.emotion-2 svg {
  width: 24px;
  min-width: 24px;
  height: 24px;
  min-height: 24px;
  margin: 0.25rem;
}

.emotion-2:hover {
  background-color: transparent;
}

.emotion-2:active {
  background-color: transparent;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-6 {
  background-color: #FFFFFF;
  color: #323845;
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  border-color: #E0E2E7;
  padding: 0rem;
}

.emotion-0 {
  min-width: 24px;
  min-height: 24px;
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(90deg);
  -moz-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition: -webkit-transform 150ms ease;
  transition: transform 150ms ease;
}

<div
    class="emotion-6"
  >
    <div
      class="emotion-5"
    >
      <header
        class="emotion-4"
      >
        <button
          aria-controls="mocked-id"
          aria-expanded="false"
          class="emotion-2"
          type="button"
        >
          <div
            class="emotion-1"
          >
            <svg
              class="emotion-0"
              fill="none"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 14L12 10L8 14"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
            Collapsible Header
          </div>
        </button>
        <div
          class="emotion-3"
        />
      </header>
    </div>
  </div>
</DocumentFragment>
`;
